hyperspy/signal.py

```diff
@@ -3111,6 +3112,79 @@ def crop(self, axis, start=None, end=None, convert_units=False):
         if convert_units:
             self.axes_manager.convert_units(axis)
 
+    def interpolate_on_axis(self,
+                            new_axis,
+                            index_in_axes_manager=0,
+                            inplace=False,
+                            set_navigate=True,
+                            degree=1):
+        """Replaces the axis specified by index_in_axes_manager with the provided new_axis
+        and interpolates data accordingly.
+
+        Parameters
+        ----------
+        new_axis : UniformDataAxis, DataAxis or FunctionalDataAxis
+            Axis which replaces the axis specified with replace_index.
+            If this new axis exceeds the range of the old axis,
+            a warning is raised that the data will be extrapolated.
+
+        index_in_axes_manager : int, default=0
+            Specifies the axis which will be replaced using the index of the
+            axis in the `axes_manager`.
+
+        inplace : bool, default=False
+            If ``True`` the data of `self` is replaced by the result and
+            the axis is changed inplace. Otherwise `self` is not changed
+            and a new signal with the changes incorporated is returned.
+
+        set_navigate : bool, default=True
+            If ``True``, sets the `navigate` attribute of `new_axis` to the value given
+            by the old axis.
```

Comment:
Is it not redundant with `new_axis`, as this should be already defined there? Except if there is a use case, I would remove it to simplify?

Reply:
Yes, it is redundant. I think the usual use case would be to keep the same `navigate` attribute during the exchange. While testing the function out, I sometimes forgot to specify `navigate` for `new_axis` or I wanted to use one axis to replace both signal and navigation axes (this is most likely not a real use case). That is why I added this parameter. But thinking about it again, changing `navigate` inside the function can be confusing. So I added a warning instead (if `navigate` of `new_axis` and `axis` do not match).

Reply:
I am wondering, if this should raise an error instead of a warning? We would expect the consumer of the function to provide a suitable axis.
The new axis is set using `hyperspy.axes.AxesManager.set_axis`, which doesn't call `hyperspy.axes.AxesManager._update_attributes` and therefore, if the `navigate` attribute changes, this will most likely make the state of the AxesManager inconsistent.

Reply:
I thought that I leave the possibility open to exchange signal/navigation axis, although it is not really intended by this method. If this should not be done anyways, because the state of AxesManager gets inconsistent. Then I change it to an error. Or another possibility would be to change `navigate` of `new_axis` if it differs and just throw a warning.

Reply:
I would keep it focused on its primary purpose and if there an ambiguity, prefer to have an error rather a warning to force the user to adjust as it is expected to be, which should be fairly trivial for the user to fix. @jlaehne, what do you think?

Reply:
> The new axis is set using `hyperspy.axes.AxesManager.set_axis`, which doesn't call `hyperspy.axes.AxesManager._update_attributes` and therefore, if the `navigate` attribute changes, this will most likely make the state of the AxesManager inconsistent.

Actually, this is incorrect, the `_update_attributes` is being called when calling `set_axis` because of a traits callback.

Reply:
> I would keep it focused on its primary purpose and if there an ambiguity, prefer to have an error rather a warning to force the user to adjust as it is expected to be, which should be fairly trivial for the user to fix. @jlaehne, what do you think?

The use cases I have in mind don't need to change the `navigate` attribute. So I'm fine with throwing an error to force users to change the attribute explicitly should they desire to do so.

Reply:
I changed it to an error now.